lib/adapter.js

```diff
@@ -70,24 +76,47 @@ karma.start = () => {
                 }
         }
 
-        if (!workerUrl) {
-                throw Error('[karma-mocha-webworker] Unable to find url for worker.js in __karma__.files');
-        }
-
         if (!mochaUrl) {
                 throw Error('[karma-mocha-webworker] Unable to find url for mocha/mocha.js in __karma__.files');
         }
 
-        const worker = new Worker(workerUrl);
+        let worker;
+        let workerPort;
+
+        if (workerType === 'Worker') {
+                if (!workerUrl) {
+                        throw Error('[karma-mocha-webworker] Unable to find url for worker.js in __karma__.files');
+                }
+
+                worker = new Worker(workerUrl);
+                workerPort = worker;
+        }
+        else if (workerType === 'SharedWorker') {
+                if (!SharedWorker) {
+                        throw new Error('[karma-mocha-webworker] There is no support for shared workers');
+                }
+
+                if (!sharedWorkerUrl) {
+                        throw Error('[karma-mocha-webworker] Unable to find url for shared-worker.js in __karma__.files');
+                }
+
+                worker = new SharedWorker(sharedWorkerUrl);
+                workerPort = worker.port;
+        }
+        else {
+                throw new Error('[karma-mocha-webworker] Unexpected Worker Class ' + workerType);
```

Comment:
Users might not understand what this error means (without `grep`ing our source). We should clarify that their configuration is incorrect. Maybe something like?:

```javascript
throw new Error(
    '[karma-mocha-webworker] Invalid karma configuration value for "client.mochaWebWorker.worker". ' + 
    'Expected "Worker" or "SharedWorker" instead of "' + workerType + '"'
);
```